### ISSUE TITLE
[i18n] add option to disable duplicate id check

### ISSUE THREAD

**Issue by marcuskrahl** [angular-eslint/angular-eslint#867]
**Description and reproduction of the issue**

Currently the i18n rule will always report errors when custom ids are used multiple times.

However there may be cases where this is not desired, for example when using common labels throughout the application

```html
<button type="button" i18n="@@common.cancel-label">Cancel</button>

<!-- later in the same file-->

<button type="button" i18n="@@common.cancel-label">Cancel</button>
```

Angular already reports duplicate ids which do not have the same content. So an option should be introduced to disable the duplicate id check.

```JSON
{
  "rules": {
    "@angular-eslint/template/i18n": ["warn"]
  }
}
```

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `13.0.1` |
| `@angular-eslint/template-parser`         | `13.0.1` |
| `@typescript-eslint/parser`               | `4.33.0` |
| `ESLint`                                  | `7.32.0` |
| `node`                                    | `14.18.2` |

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.
